src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5439,10 +5438,14 @@ void Unit::RemoveAreaTrigger(AuraEffect const* aurEff)
     }
 }
 
-void Unit::RemoveAllAreaTriggers()
+void Unit::RemoveAllAreaTriggers(bool force /*= true*/)
 {
```

Comment:
this was called at JustDied or Reset?

Reply:
oh, wait, weren't areatriggers despawned at death by default? in order to prevent that, you'd have to use JustDied override, being empty, however, this flag could just prevent that small workaround

Reply:
Notice "force" arg. Currently at despawning isnt handled in a generic case besides RemoveFromWorld. We could add that part to BossAI::JustDied and BossAI::EnterEvadeMode however